parastell/utils.py

```diff
@@ -202,3 +207,119 @@ def smooth_matrix(matrix, steps, sigma):
+        """
+        self.mb.load_file(filename)
+
+    def get_max_ids(self):
+        """Get the maximum id in the DAGMC model for each entity type.
+
+        returns:
+            max_ids (dict): {entity dimension (int): max_id (int)}
+        """
+        # dim 0: vertex
+        # dim 1: edge
+        # dim 2: surface
+        # dim 3: volume
+        # dim 4: group
+        max_ids = {dim: 0 for dim in range(5)}
+        root_set = self.mb.get_root_set()
+        geom_sets = self.mb.get_entities_by_type_and_tag(
+            root_set, types.MBENTITYSET, self.geom_dim_tag, [None]
+        )
+
+        for entity_set in geom_sets:
+            dim = self.mb.tag_get_data(self.geom_dim_tag, entity_set)[0][0]
+            gid = self.mb.tag_get_data(self.global_id_tag, entity_set)[0][0]
+            max_ids[dim] = max(max_ids[dim], gid)
+
+        return max_ids
+
+    def reassign_global_ids(self, offsets):
+        """Renumber entities, starting from the offset value for that
+            dimension.
+
+        Arguments:
+            offsets (dict): {entity dimension (int): id_offset (int)}
+        """
+        root_set = self.mb.get_root_set()
+        geom_sets = self.mb.get_entities_by_type_and_tag(
+            root_set, types.MBENTITYSET, self.geom_dim_tag, [None]
+        )
+
+        for entity_set in geom_sets:
+            dim = self.mb.tag_get_data(self.geom_dim_tag, entity_set)[0][0]
+            current_gid = self.mb.tag_get_data(self.global_id_tag, entity_set)[
+                0
+            ][0]
+            self.mb.tag_set_data(
+                self.global_id_tag,
+                entity_set,
+                current_gid + offsets.get(dim, 0),
+            )
+
+    def write_file(self, filename):
+        """Save to file.
+
+        Arguments:
+            filename (str): Path to save the file to. All filetypes supported
+                by MOAB are available. Write to h5m for use with DAGMC.
+        """
+        self.mb.write_file(filename)
+
+
+def merge_dagmc_files(models_to_merge):
+    """Takes a list of DAGMC models, and renumbers entity ids such that they
+    will no longer clash, allowing the models to be loaded into the same
+    PyMOAB core instance, and saved to a single model.
+
+    Arguments:
+        models_to_merge (list of PyMOAB core): List of DAGMC models to be
+            merged.
+    Returns:
+        merged_model (dagmc.DAGModel): Single DAGMC model containing each
+            individual model.
+    """
+    temp_files = []
+    max_ids = {dim: 0 for dim in range(5)}
+    for model in models_to_merge:
+        merger = DAGMCMerger(model)
+        merger.reassign_global_ids(max_ids)
+        max_ids = merger.get_max_ids()
+        with tempfile.NamedTemporaryFile(
+            delete=False, suffix=".h5m"
+        ) as temp_file:
+            temp_filename = temp_file.name
+        merger.write_file(temp_filename)
+        temp_files.append(temp_filename)
+
+    merged_mbc = core.Core()
```

Comment:
I think you can instantiate this MOAB core at the beginning of the method and just reuse the same temp_file, loading it into this MOAB core each time.